### ISSUE TITLE
SV - Unregistered user can submit Upcoming Milestone

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-09-Gitcoin-judging#842]
SV

high

# Unregistered user can submit Upcoming Milestone

In RFPSimpleStrategy , recipients can call  `submitUpcomingMilestone` with their proof of work. The docs mention that the caller must be an `acceptedRecipientId` and should be a `profile` member . 

They do by checking through this:-

https://github.com/sherlock-audit/2023-09-Gitcoin/blob/main/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L255-L257

But , it does not check the case where, 
`acceptedRecipientId` == msg.sender && !_isProfileMember(acceptedRecipientId, msg.sender)

## Vulnerability Detail

Paste the below code in RFPSimpleStrategy.t.sol
Run `forge test --mt test_poc_malicious_user_can_front_run_and_set_upcoming_milestone`
```solidity
 function test_poc_malicious_user_can_front_run_and_set_upcoming_milestone() public {
        address recipient = makeAddr("malicious_recipient");
        Metadata memory metadata = Metadata({protocol: 1, pointer: "metadata"});

        bytes memory data = abi.encode(recipient, false, 1e18, metadata);
        vm.prank(recipient);
        address recipientId = allo().registerRecipient(poolId, data);
        __setMilestones();

        emit Allocated(recipientId, 1e18, NATIVE, address(pool_admin()));

        data = abi.encode(recipient, false, 1e18, metadata);

        vm.prank(pool_admin());
        allo().allocate(poolId, abi.encode(recipientId));
        vm.expectEmit();
        emit MilstoneSubmitted(0);
       
        vm.prank(recipient);
        strategy.submitUpcomingMilestone(Metadata({protocol: 1, pointer: "metadata"}));
        
        RFPSimpleStrategy.Milestone memory milestone = strategy.getMilestone(0);
        assertEq(uint8(milestone.milestoneStatus), uint8(IStrategy.Status.Pending)); 
    }
```
Here, first `registerRecipient` is called, we assume , `useRegistryAnchor` is false for the pool. 
https://github.com/sherlock-audit/2023-09-Gitcoin/blob/main/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L327-L346

Hence , we jump to else block.
As `registryAnchor` == address(0), `isUsingRegistryAnchor` will be false.
Hence , this check passes
```solidity
            if (isUsingRegistryAnchor && !_isProfileMember(recipientId, _sender)) revert UNAUTHORIZED();
```

Now even though recipient was not a profile member, they can call `submitUpcomingMilestone` . They will pass the check as they are acceptedRecipientId but not a profile Member.
https://github.com/sherlock-audit/2023-09-Gitcoin/blob/main/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L255

## Impact
Recipient which are not even a profile member can set milestones . Now they can deceive everyone in the protocol who believe they are registered as a profile member.
## Code Snippet
https://github.com/sherlock-audit/2023-09-Gitcoin/blob/main/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L253-L271
## Tool used

Manual Review

## Recommendation
Ensure that only user should be a profile member to be able to call `submitUpcomingMilestone`

Duplicate of #259

**Comment by viraj124** [comment]
Escalate- the sponsor `<jaxcoder />` confirmed this issue on a discord chat with us and we think this is a valid issue so we would like to escalate this 
Thanks

**Comment by sherlock-admin2** [comment]
> Escalate- the sponsor `<jaxcoder />` confirmed this issue on a discord chat with us and we think this is a valid issue so we would like to escalate this 
> Thanks 

The escalation could not be created because you are not exceeding the escalation threshold.

You can view the required number of additional valid issues/judging contest payouts in your Profile page,
in the [Sherlock webapp](https://app.sherlock.xyz/audits/).

**Comment by viraj124** [comment]
Escalate

the sponsor `<jaxcoder />` confirmed this issue on a discord chat with us this was his response
```
I think it should be or not and
maybe even two different if statements
```
and we think this is a valid issue so we would like to escalate this as this allows milestone creation to be unauthorized and can lead to issues as milestones play an important role in a project meeting it's funding goals
Thanks

**Comment by sherlock-admin2** [comment]
> Escalate
> 
> the sponsor 1<jaxcoder />1 confirmed this issue on a discord chat with us and we think this is a valid issue so we would like to escalate this as this allows milestone creation to be unauthorized and can lead to issues as milestones play an important role in a project meeting it's funding goals
> Thanks

The escalation could not be created because you are not exceeding the escalation threshold.

You can view the required number of additional valid issues/judging contest payouts in your Profile page,
in the [Sherlock webapp](https://app.sherlock.xyz/audits/).

**Comment by thelostone-mc** [comment]
Agreed. This is an issue for the comment.
The code works as expected. We have to update the documentation

**Comment by thelostone-mc** [comment]
The title of the issue is not valid. 
An unregistered user cannot submit upcoming milestone.
The natspec comment is incorrect but this is already called out and fixed in
https://github.com/sherlock-audit/2023-09-Gitcoin-judging/issues/259

Best case scenario: this is a duplicate of that but the title of the issue is invalid.
Defer to sherlock on duplication status

**Comment by viraj124** [comment]
@thelostone-mc can we get this included as one of the duplicates for #259 ?
